### PR TITLE
Device agent docs

### PR DESCRIPTION
**Pull request by bryfox**
This adds initial documentation for the Foxglove Agent.

### CODE REVIEW

File: docs/12-on-device-agents/1-installation.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+Copy the binary to your device, and run with the provided command-line arguments.
+
+<p class="Debug">suggest some sort of process management tool here? init.d or whatever</p>
+
+<p class="Debug">Note: code block below is TBD — what are the minimum set of flags?</p>
+
+```sh
+./foxglove-device-agent \
+    --device-token ${DEVICE_TOKEN_SECRET} \
+    --storage-root "./my-directory"
```
Comment (wkalt) [inline docs/12-on-device-agents/1-installation.md:47]: wyatt to update with minimal config

File: docs/12-on-device-agents/0-introduction.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+title: Introduction
+description: An On-Device Agent is a Foxglove process that runs on your robot. It's designed to monitor the robot's filesystem for logs and notify the Foxglove control plane when recordings are available for import. As with a Edge Site, you control which recordings are imported and when by selecting them for import.
+---
+
+import AccountRequiredHeader from "../../src/components/docs/icons/AccountRequiredHeader";
+
+<AccountRequiredHeader badgeText="Requires Enterprise plan" />
+
+An On-Device Agent is a Foxglove process that runs on your robot. It's designed to monitor the robot's filesystem for logs and notify the Foxglove control plane when recordings are available for import. As with a Edge Site, you control which recordings are imported and when by selecting them for import.
```
Comment (james-rms) [inline docs/12-on-device-agents/0-introduction.md:10]:
```suggestion
An On-Device Agent is a Foxglove process that runs on your robot. It's designed to monitor the robot's filesystem for new recordings and notify the Foxglove control plane when they are available for import. As with a Edge Site, you use Foxglove to control which recordings are imported and when.
```

File: docs/12-on-device-agents/1-installation.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  --azure-storage-account-name
+  --azure-storage-service-url
+```
+-->
+
+### Confirm running
+
+If misconfigured, the agent will log errors and exit with a nonzero status.
+
+For systemd, `sysctl status` will report that the agent is running.
```
Comment (james-rms) [inline docs/12-on-device-agents/1-installation.md:101]:
```suggestion
For systemd, `sysctl status <service unit name>` will report that the agent is running.
```

File: docs/12-on-device-agents/4-monitoring.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+---
+title: Monitoring
+description: Monitor your On-Device Agent.
+---
+
+Monitor your On-Device Agent.
+
+The agent is capable of publishing Prometheus metrics. To enable this, supply the desired port of the prometheus server using `--prometheus-port`.
```
Comment (james-rms) [inline docs/12-on-device-agents/4-monitoring.md:8]: what could realistically consume this? From the likely user's POV, this is not useful information.

File: docs/12-on-device-agents/1-installation.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+curl https://example.com
+```
+
+<span class="Debug">command tbd</span>
+
+### Run the agent
+
+Copy the binary to your device, and run with the provided command-line arguments.
+
+<p class="Debug">[todo] wyatt to create a systemd file</p>
```
Comment (james-rms) [inline docs/12-on-device-agents/1-installation.md:50]: possibly also create a user account to run this agent

File: docs/12-foxglove-agent/1-installation.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+Create a device from the [Devices page](https://console.foxglove.dev/devices), or select one that you've already created.
+
+An admin for your Foxglove organization can then go to that device's details page, navigate to the "Device Tokens" tab, and generate a secret token. This token will be displayed only once, so make sure you take note of it and supply it to the agent during configuration.
+
+An admin can also create a device token via the Foxglove API. On the [API key settings page](https://console.foxglove.dev/settings/apikeys), create an API key with the appropriate `deviceToken` capabilities (list, create, etc), and use it to create device tokens via the [device tokens API endpoint](/api#tag/Device-Tokens) <span class="Debug">(verify link once published)</span>.
+
+### Download and upgrade the Foxglove Agent package
+
+Download the latest `foxglove-agent` package for your architecture from the [releases page](https://github.com/foxglove/agent/releases). Install with a package manager like `dpkg`:
```
Comment (james-rms) [inline docs/12-foxglove-agent/1-installation.md:29]: `dpkg` is the only package manager that will install these files.
```suggestion
Download the latest `foxglove-agent` package for your architecture from the [releases page](https://github.com/foxglove/agent/releases). Install with `dpkg`:
```

File: docs/12-foxglove-agent/1-installation.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+<AccountRequiredHeader badgeText="Requires Enterprise plan" />
+
+Install and configure the Foxglove Agent.
+
+### Prerequisites
+
+- Debian-based Linux distribution (ARM64 and AMD64 architectures supported)
+- HTTPS connectivity to `api.foxglove.dev` (stable connection not required)
+- Filesystem that supports `fsnotify`
```
Comment (james-rms) [inline docs/12-foxglove-agent/1-installation.md:16]: a robot running linux will likely have several filesystems, only the one hosting the `STORAGE_ROOT` requires `fsnotify` support.

Comment (james-rms) [reply docs/12-foxglove-agent/1-installation.md:16]:
```suggestion
- The filesystem hosting your recordings storage directory must support`fsnotify`
```

File: docs/12-foxglove-agent/1-installation.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+- Directory for storing local state files
+
+### Create a device and device token
+
+Create a device from the [Devices page](https://console.foxglove.dev/devices), or select one that you've already created.
+
+An admin for your Foxglove organization can then go to that device's details page, navigate to the "Device Tokens" tab, and generate a secret token. This token will be displayed only once, and is required for agent configuration.
+
+An admin can also create a device token via the Foxglove API. On the [API key settings page](https://console.foxglove.dev/settings/apikeys), create an API key with the appropriate `deviceToken` capabilities (list, create, etc), and use it to create device tokens via the [device tokens API endpoint](/api#tag/Device-Tokens).
```
Comment (james-rms) [inline docs/12-foxglove-agent/1-installation.md:25]: Section is available now at `https://docs.foxglove.dev/api#tag/Device-Tokens`.

**Pull request merged**